models/user.go

```diff
@@ -957,6 +963,26 @@ func updateUser(e Engine, u *User) error {
 
 // UpdateUser updates user's information.
 func UpdateUser(u *User) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	u.Email = strings.ToLower(u.Email)
+	isExist, err := sess.
+		Where("email=?", u.Email).
+		Get(new(User))
+	if err != nil {
+		return err
+	} else if isExist {
```

Comment:
But if the user is himself?

Reply:
Yup, I didn't consider that scenario. Fixed!